model/user.go

```diff
@@ -1,6 +1,78 @@
 package model
 
+import (
+	"errors"
+	"log"
+
+	helper "web/helpers"
+
+	"gorm.io/gorm"
+)
+
+var DB *gorm.DB
+
 type User struct {
-	UserID string `json:"user_id"`
-	Name   string `json:"name"`
+	gorm.Model
+	Email    string `gorm:"size64, index"`
+	Username string `gorm:"size64, index"`
+	Password string `gorm:"size255"`
+}
+
+func CheckUserExistance(email string) bool {
+	var user User
+
+	if err := DB.Where("email = ?", email).First(&user).Error; err == nil {
+		return true
+	}
+
+	if err := DB.Where("username = ?", email).First(&user).Error; err == nil {
+		return true
+	}
+
+	return false
+}
+
+func CreateUser(email, username, password string) *User {
+	encryptedPassword, err := helper.EncryptPassword(password)
+	if err != nil {
+		log.Fatalf("Failed to encrypt password: %v", err)
+	}
+
+	user := User{
+		Email:    email,
+		Username: username,
+		Password: encryptedPassword,
+	}
+	DB.Create(&user)
+	if user.ID == 0 {
+		log.Fatalf("Failed to create user: %v", user)
+	}
+
+	log.Printf("User created: %v", user)
+	return &user
+}
+
+func CheckPasswordMatch(username, password string) (*User, error) {
+	var user User
+	DB.Where("username = ?", username).First(&user)
+	if user.ID == 0 {
+		return nil, errors.New("invalid username")
+	}
+
+	err := helper.CheckPasswordMatch(user.Password, password)
+	if err != nil {
+		return nil, errors.New("invalid password")
+	}
+	return &user, nil
```

Comment:
For me, the spacing would be great here
```suggestion
	}

	return &user, nil
```

Reply:
This comment is left?